src/wp-content/themes/twentytwentyone/assets/js/interactivity.js

```diff
@@ -0,0 +1,206 @@
+            const ctx = getContext();
+
+            // Check if this was a `.sub-menu-toggle` click.
+            const subMenuToggle = checkClass( event.target, 'sub-menu-toggle' );
+            if ( subMenuToggle ) {
+                if ( ctx.activeSubmenu === subMenuToggle ) {
+                    ctx.activeSubmenu = null;
+                } else {
+                    ctx.activeSubmenu = subMenuToggle;
+                }
+                return;
+            }
+
+            // Otherwise, check if this was an anchor link click.
+            if ( ! event.target.hash ) {
+                return;
+            }
+
+            actions.closePrimaryMenu();
+
+            // Wait 550 and scroll to the anchor.
+            setTimeout( () => {
+                var anchor = document.getElementById( event.target.hash.slice( 1 ) );
+                if ( anchor ) {
+                    anchor.scrollIntoView();
+                }
+            }, 550 );
+        },
+    },
+    callbacks: {
+        determineFocusableElements: () => {
+            if ( ! state.isPrimaryMenuOpen ) {
+                return;
+            }
+
+            const ctx = getContext();
+            const { ref } = getElement();
+			const elements = ref.querySelectorAll( 'input, a, button' );
+
+            ctx.firstFocusable = elements[ 0 ];
+            ctx.lastFocusable = elements[ elements.length - 1 ];
+        },
+
+        refreshSubmenus: () => {
+            const ctx = getContext();
+            const { ref } = getElement();
+			const elements = ref.querySelectorAll( '.sub-menu-toggle' );
+            elements.forEach( ( subMenuToggle ) => {
+                if ( ctx.activeSubmenu === subMenuToggle ) {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'true' );
+                } else {
+                    subMenuToggle.setAttribute( 'aria-expanded', 'false' );
+                }
+            } );
+        },
+
+        makeIframesResponsive: () => {
+            const { ref } = getElement();
+
+            ref.querySelectorAll( 'iframe' ).forEach( function( iframe ) {
+                // Only continue if the iframe has a width & height defined.
+                if ( iframe.width && iframe.height ) {
+                    // Calculate the proportion/ratio based on the width & height.
+                    proportion = parseFloat( iframe.width ) / parseFloat( iframe.height );
+                    // Get the parent element's width.
+                    parentWidth = parseFloat( window.getComputedStyle( iframe.parentElement, null ).width.replace( 'px', '' ) );
+                    // Set the max-width & height.
+                    iframe.style.maxWidth = '100%';
+                    iframe.style.maxHeight = Math.round( parentWidth / proportion ).toString() + 'px';
+                }
+            } );
+        },
+
+        updateWindowWidthOnResize: () => {
+            // The following may be needed here since we can't use `data-wp-on--resize`?
```

Comment:
Similar answer to the `scroll` global event, this should be fine as a `data-wp-init` for now.

Reply:
Sounds good. Replied in https://github.com/WordPress/wordpress-develop/pull/5795#discussion_r1444915934 about potentially making this more intuitive/declarative.